### ISSUE TITLE
Find-Module AllVersions switch doesn't work on Linux/Ubuntu 18.04 with 'C' Locale

### ISSUE THREAD

**Issue by splatteredbits** [PowerShell/PowerShellGetv2#399]
Steps to reproduce
------------------

```powershell
Find-Module PowerShellGet -AllVersions
```

Expected behavior
-----------------
This is what I get on Windows:

```none
Version Name          Repository Description
------- ----          ---------- -----------
2.0.3   PowerShellGet PSGallery  PowerShell module with commands for discovering, installing, updating and publishing the PowerShell artifacts like Modules, DSC Resources, Role Capabilities and Scripts.
2.0.1   PowerShellGet PSGallery  PowerShell module with commands for discovering, installing, updating and publishing the PowerShell artifacts like Modules, DSC Resources, Role Capabilities and Scripts.
2.0.0   PowerShellGet PSGallery  PowerShell module with commands for discovering, installing, updating and publishing the PowerShell artifacts like Modules, DSC Resources, Role Capabilities and Scripts.
1.6.7   PowerShellGet PSGallery  PowerShell module with commands for discovering, installing, updating and publishing the PowerShell artifacts like Modules, DSC Resources, Role Capabilities and Scripts.
1.6.6   PowerShellGet PSGallery  PowerShell module with commands for discovering, installing, updating and publishing the PowerShell artifacts like Modules, DSC Resources, Role Capabilities and Scripts.
1.6.5   PowerShellGet PSGallery  PowerShell module with commands for discovering, installing, updating and publishing the PowerShell artifacts like Modules, DSC Resources, Role Capabilities and Scripts.
1.6.0   PowerShellGet PSGallery  PowerShell module with commands for discovering, installing, updating and publishing the PowerShell artifacts like Modules, DSC Resources, Role Capabilities and Scripts.
1.5.0.0 PowerShellGet PSGallery  PowerShell module with commands for discovering, installing, updating and publishing the PowerShell artifacts like Modules, DSC Resources, Role Capabilities and Scripts.
1.1.3.2 PowerShellGet PSGallery  PowerShell module with commands for discovering, installing, updating and publishing the PowerShell artifacts like Modules, DSC Resources, Role Capabilities and Scripts.
1.1.3.1 PowerShellGet PSGallery  PowerShell module with commands for discovering, installing, updating and publishing the PowerShell artifacts like Modules, DSC Resources, Role Capabilities and Scripts.
1.1.2.0 PowerShellGet PSGallery  PowerShell module with commands for discovering, installing, updating and publishing the PowerShell artifacts like Modules, DSC Resources, Role Capabilities and Scripts.
1.1.1.0 PowerShellGet PSGallery  PowerShell module with commands for discovering, installing, updating and publishing the PowerShell artifacts like Modules, DSC Resources, Role Capabilities and Scripts.
1.1.0.0 PowerShellGet PSGallery  PowerShell module with commands for discovering, installing, updating and publishing the PowerShell artifacts like Modules, DSC Resources, Role Capabilities and Scripts.
```

Actual behavior
---------------
I only get the latest version on Linux:
```none
Version Name          Repository Description
------- ----          ---------- -----------
2.0.3   PowerShellGet PSGallery  PowerShell module with commands for discovering, installing, updating and publishing the PowerShell artifacts like Modules, DSC Resources, Role Capabilities and Scripts.
```

Environment data
----------------

```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      6.1.1
PSEdition                      Core
GitCommitId                    6.1.1
OS                             Linux 4.4.0-17134-Microsoft #471-Microsoft Fri Dec 07 20:04:00 PST 2018
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

```powershell
> Get-Module PowerShellGet,PackageManagement
ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     1.2.4      PackageManagement                   {Find-Package, Find-PackageProvider, Get-Package, Get-PackageProvider...}
Script     2.0.3      PowerShellGet                       {Find-Command, Find-DscResource, Find-Module, Find-RoleCapability...}
```

```powershell
> Get-PackageProvider -ListAvailable
Name                     Version          DynamicOptions
----                     -------          --------------
NuGet                    3.0.0.1          Destination, ExcludeVersion, Scope, SkipDependencies, Headers, FilterOnTag, Contains, AllowPrereleaseVersions, ConfigFile, SkipValidate
PowerShellGet            1.6.7.0          PackageManagementProvider, Type, Scope, AllowClobber, SkipPublisherCheck, InstallUpdate, NoPathUpdate, AllowPrereleaseVersions, Filter, Tag, Includes, DscResource, RoleCapability, Command, Ac...

```

**Comment by edyoung** [comment]
I can reproduce this, but only in Windows Subsystem for Linux. Tried this with 'real' Ubuntu on the same version of Ubuntu, Powershell, Package Management and PowerShellGet, and it worked fine. In WSL only the first version shows up. @splatteredbits I assume you are also using WSL but let me know if that's inaccurate.

**Comment by splatteredbits** [comment]
Correct. This is only happening on Ubuntu running under the Windows Subsystem for Linux.

**Comment by splatteredbits** [comment]
This also appears to be happening on the Ubuntu VMs being used by AppVeyor, our build tool. I confirmed with them that their images are, in fact, standalone Ubuntu VMs, and not running on WSL.

**Comment by edyoung** [comment]
It looks like this is also resolved by the same change in PowerShellGet 2.0.4 - I can no longer repro in WSL after the update. Please reactivate if you disagree.

**Comment by splatteredbits** [comment]
This still repros for me on WSL with PowerShellGet 2.0.4 and both PackageManagement versions 1.3.0 and 1.2.4.

**Comment by edyoung** [comment]
Thanks for the correction- looks like I mixed up the effects of 2 separate tests I tried. As you say, this issue is NOT resolved by updating to PowerShellGet 2.0.4, but it is worked-around by changing the locale/LANG variable. In WSL LANG=C.UTF-8 by default. 

```
bash> export LANG=en_US.UTF-8
bash> pwsh
pwsh> Find-Module PowerShellGet -AllVersions
```
Works in my testing. There will be some part of the code doing a language-sensitive string comparison where it should not which is leading to this issue.

**Comment by splatteredbits** [comment]
Thanks for the workaround. Setting that environment variables fixes my issue. Hopefully a more permanent fix can be found.

**Comment by edyoung** [comment]
See https://github.com/OneGet/oneget/pull/451 (bug is in oneget, not powershellget)

**Comment by edyoung** [comment]
fixed in 2.1.4/packagemanagement 1.4
